sympy/sets/fancysets.py

```diff
@@ -443,3 +445,309 @@ def _sup(self):
+    >>> from sympy import S, I, Union
+    >>> a = Interval(2, 3)
+    >>> b = Interval(4, 6)
+    >>> c = Interval(1, 8)
+    >>> c1 = ComplexPlane(a*b)  # Rectangular Form
+    >>> c1
+    ComplexPlane(Lambda((x, y), x + I*y), [2, 3] x [4, 6])
+
+    * c1 represents the rectangular region in complex plane
+      surrounded by the coordinates (2, 4), (3, 4), (3, 6) and
+      (2, 6), of the four vertices.
+
+    >>> c2 = ComplexPlane(Union(a*b, b*c))
+    >>> c2
+    ComplexPlane(Lambda((x, y), x + I*y),
+                 [2, 3] x [4, 6] U [4, 6] x [1, 8])
+
+    * c2 represents the Union of two rectangular regions in complex
+      plane. One of them surrounded by the coordinates of c1 and
+      other surrounded by the coordinates (4, 1), (6, 1), (6, 8) and
+      (4, 8).
+
+    >>> 2.5 + 4.5*I in c1
+    True
+    >>> 2.5 + 6.5*I in c1
+    False
+
+    >>> r = Interval(0, 1)
+    >>> theta = Interval(0, 2*S.Pi)
+    >>> c2 = ComplexPlane(r*theta, polar=True)  # Polar Form
+    >>> c2  # unit Disk
+    ComplexPlane(Lambda((r, theta), r*(I*sin(theta) + cos(theta))),
+                 [0, 1] x [0, 2*pi])
+
+    * c2 represents the region in complex plane inside the
+      Unit Disk centered at the origin.
+
+    >>> 0.5 + 0.5*I in c2
+    True
+    >>> 1 + 2*I in c2
+    False
+
+    >>> unit_disk = ComplexPlane(Interval(0, 1)*Interval(0, 2*S.Pi), polar=True)
+    >>> upper_half_unit_disk = ComplexPlane(Interval(0, 1)*Interval(0, S.Pi), polar=True)
+    >>> intersection = unit_disk.intersect(upper_half_unit_disk)
+    >>> intersection
+    ComplexPlane(Lambda((r, theta), r*(I*sin(theta) + cos(theta))), [0, 1] x [0, pi])
+    >>> intersection == upper_half_unit_disk
+    True
+
+    See Also
+    ========
+
+    Reals
+
+    """
+    is_ComplexPlane = True
+
+    def __new__(cls, sets, polar=False):
+        from sympy import symbols
+
+        x, y, r, theta = symbols('x, y, r, theta')
+        I = S.ImaginaryUnit
+
+        # Rectangular Form
+        if polar is False:
+            obj = ImageSet.__new__(cls, Lambda((x, y), x + I*y),
+                                   sets)
+
+        # Polar Form
+        elif polar is True:
+            from sympy import cos, sin
+            obj = ImageSet.__new__(cls, Lambda((r, theta),
+                                   r*(cos(theta) + I*sin(theta))),
+                                   sets)
+        return obj
+
+    @property
+    def sets(self):
+        """
+        Return raw input sets to the self.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> c = Interval(1, 7)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.sets
+        [2, 3] x [4, 5]
+        >>> C2 = ComplexPlane(Union(a*b, b*c))
+        >>> C2.sets
+        [2, 3] x [4, 5] U [4, 5] x [1, 7]
+
+        """
+        return self.args[1]
+
+    @property
+    def psets(self):
```

Comment:
Same as this [comment](https://github.com/sympy/sympy/pull/9463/files#r32110887).
Increases readability and simplifies code further in other methods.